discourse/discourse.php

```diff
@@ -0,0 +1,354 @@
+<?php
+
+/**
+ * Name: Discourse Mail Connector
+ * Description: Improves mails from Discourse in mailing list mode
+ * Version: 0.1
+ * Author: Michael Vogel <http://pirati.ca/profile/heluecht>
+ *
+ */
+//use DOMDocument;
+//use DOMXPath;
+use Friendica\App;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Logger;
+use Friendica\Core\PConfig;
+use Friendica\Core\Renderer;
+use Friendica\Core\Protocol;
+use Friendica\Database\DBA;
+use Friendica\Model\Contact;
+use Friendica\Util\XML;
+use Friendica\Content\Text\Markdown;
+use Friendica\Util\Network;
+use Friendica\Util\Strings;
+Use Friendica\Util\DateTimeFormat;
+
+/* Todo:
+ * - Obtaining API tokens to be able to read non public posts as well
+ * - Handling duplicates (possibly using some non visible marker)
+ * - Fetching missing posts
+ * - Fetch topic information
+ * - Support mail free mode when write tokens are available
+ * - Fix incomplete (relative) links (hosts are missing)
+*/
+
+function discourse_install()
+{
+	Hook::register('email_getmessage',        __FILE__, 'discourse_email_getmessage');
+	Hook::register('connector_settings',      __FILE__, 'discourse_settings');
+	Hook::register('connector_settings_post', __FILE__, 'discourse_settings_post');
+}
+
+function discourse_uninstall()
```

Comment:
If your `*_uninstall` function doesn't have anything else than hook unregistration, you can scrap it entirely, the addon uninstalling process now removes all hooks related to this addon by default. This addresses obsolete hook registration as well without having to name them explicitly in `*_uninstall`.